### PR TITLE
Do not apply RateLimiter when using Agent sampling rates

### PR DESCRIPTION
**Pull request by colin-higgins**
Expected behavior:
 - `_dd.limit_psr` is always set when `_dd.rule_psr` is set
 - `_dd.limit_psr` is never set when `_dd.agent_psr` is set
 - Do not apply the Rate Limiter when sampling is from the agent.

### CODE REVIEW

File: src/Datadog.Trace/Configuration/TracerSettings.cs [hunk 1]
```diff
@@ -94,17 +94,7 @@ public TracerSettings(IConfigurationSource source)
...
-                // Ensure our flag for the rate limiter is enabled
-                RuleBasedSampler.OptInTracingWithoutLimits();
-            }
-            else
-            {
-                maxTracesPerSecond = 100; // default
-            }
-
-            MaxTracesSubmittedPerSecond = maxTracesPerSecond.Value;
+            MaxTracesSubmittedPerSecond = maxTracesPerSecond ?? 100;
```
Comment (lucaspimentel) [inline src/Datadog.Trace/Configuration/TracerSettings.cs:97]: Just for consistency, can you combine this line with the previous line, like the other settings above?
(github wouldn't let me suggest)

```csharp
MaxTracesSubmittedPerSecond = source?.GetInt32(ConfigurationKeys.MaxTracesSubmittedPerSecond) ??
                              // default value
                              100;
```

File: src/Datadog.Trace/Sampling/RuleBasedSampler.cs [hunk 2]
```diff
@@ -80,14 +72,14 @@ public void RegisterRule(ISamplingRule rule)
...
-        private SamplingPriority GetSamplingPriority(Span span, float rate)
+        private SamplingPriority GetSamplingPriority(Span span, float rate, bool agentSampling)
         {
             var sample = ((span.TraceId * KnuthFactor) % TracerConstants.MaxTraceId) <= (rate * TracerConstants.MaxTraceId);
             var priority = SamplingPriority.AutoReject;
 
             if (sample)
             {
-                if (_tracingWithoutLimitsEnabled)
+                if (!agentSampling)
```
Comment (lucaspimentel) [inline src/Datadog.Trace/Sampling/RuleBasedSampler.cs:82]: If we invert this `if` condition, we can simplify as:
```csharp
if (sample)
{
	if (agentSampling)
	{
	    priority = SamplingPriority.AutoKeep;
	}
	else if (_limiter.Allowed(span))
	{
		priority = SamplingPriority.AutoKeep;
	}
}
```
which further simplifies to:
```csharp
if (sample && (agentSampling || _limiter.Allowed(span)))
{
	priority = SamplingPriority.AutoKeep;
}
```
and if we drop the local `priority`:
```csharp
var sample = ...

return sample && (agentSampling || _limiter.Allowed(span)) ?
		   SamplingPriority.AutoKeep :
		   SamplingPriority.AutoReject;
```

Comment (colin-higgins) [reply src/Datadog.Trace/Sampling/RuleBasedSampler.cs]: Is it okay with you if we keep the first suggestion for readability?

Comment (colin-higgins) [reply src/Datadog.Trace/Sampling/RuleBasedSampler.cs]: Eh, I'll change it

**Review by zacharycmontoya — APPROVED** [review]

**Review by lucaspimentel — APPROVED** [review]

**Pull request merged**